packages/superset-ui-number-format/src/factories/createSmartNumberFormatter.ts

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-magic-numbers */
+
+import { format as d3Format } from 'd3-format';
+import NumberFormatter from '../NumberFormatter';
+import NumberFormats from '../NumberFormats';
+
+export default function createSmartNumberFormatter(
+  config: {
+    description?: string;
+    signed?: boolean;
+    id?: string;
+    label?: string;
+  } = {},
+) {
+  const { description, signed = false, id, label } = config;
+  const siFormatter = d3Format(`.3~s`);
```

Comment:
we could move these outside the function so they aren't created every call? looks like they don't ever change so no reason not to?